### PR TITLE
enable build manifest caching by default

### PR DESCRIPTION
**Pull request by tomerd**
motivation: improve performance

changes:
* set the flag to true by default
* refactor the code to better reflect true meaning of flag
* warn on any error and fallback to non-cached version
* add test for disabling the cache

rdar://63981763

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
cc @aciidb0mb3r since you know this code

**Review by abertelrud — APPROVED** [review]

**Review by neonichu — APPROVED** [review]

File: Sources/Commands/SwiftTool.swift [hunk 1]
```diff
@@ -624,14 +633,15 @@ public class SwiftTool {
         return buildOp
     }
 
-    func createBuildSystem(explicitProduct: String? = nil, useBuildManifestCaching: Bool = true, buildParameters: BuildParameters? = nil) throws -> BuildSystem {
+    func createBuildSystem(explicitProduct: String? = nil, buildParameters: BuildParameters? = nil) throws -> BuildSystem {
+        print("createBuildSystem \(try self.canUseCachedBuildManifest())")
```
Comment (krzyzanowskim) [inline Sources/Commands/SwiftTool.swift:637]: @tomerd it looks like some sort of leftover debugging message, is it?

Comment (tomerd) [reply Sources/Commands/SwiftTool.swift:637]: good catch thanks

Comment (tomerd) [reply Sources/Commands/SwiftTool.swift:637]: https://github.com/apple/swift-package-manager/pull/3167

**Pull request merged**